lib/cn/csched_sp3_work.c

```diff
@@ -739,12 +739,9 @@ sp3_work(
 
     INIT_LIST_HEAD(&w->cw_rspill_link);
 
-    if (w->cw_rspill_conc) {
-        /* ensure concurrent root spills complete in order */
-        mutex_lock(&tree->ct_rspills_lock);
-        list_add_tail(&w->cw_rspill_link, &tree->ct_rspills_list);
-        mutex_unlock(&tree->ct_rspills_lock);
-    }
+    /* ensure concurrent root spills complete in order */
+    if (w->cw_rspill_conc)
+        w->cw_sgen = atomic_inc_return(&w->cw_tree->ct_sgen);
```

Comment:
Regarding the new serialization, can spill thread 2 ever spill to a node further to the right than spill thread 1?
Say T1 is working on node 3, and a bit later T2 starts on node 7.  Can T2 commit to node 7 so long as T1 is still working on node 3?

Reply:
No, it will have to wait for T1 to update node 7 (i.e. bump up node 7's sgen). Because T2 doesn't know whether T1 will have any data for node 7 or not.